sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -235,10 +243,25 @@ public void sync(CompletionListener listener) {
             LOGGER.debug("method=sync");
 
             if (throwable == null) {
+                /**
+                 * Global state is assigned to local variables so that cleanup can be called immediately.
+                 *
+                 * If bulk operations are used parse() may call into sync and clients don't wait for parseOK
```

Comment:
Plz rephrase "may call into sync" and I would add a comma: If bulk operations are used, ....